addons/analytic/models/analytic_account.py

```diff
@@ -138,22 +138,19 @@ def _read_group(self, domain, groupby=(), aggregates=(), having=(), offset=0, li
         if SPECIAL.isdisjoint(aggregates):
             return super()._read_group(domain, groupby, aggregates, having, offset, limit, order)
 
-        new_aggregates = ['id:array_agg', *(agg for agg in aggregates if agg not in SPECIAL)]
-        new_result = super()._read_group(domain, groupby, new_aggregates, having, offset, limit, order)
+        new_aggregates = ['id:recordset', *(agg for agg in aggregates if agg not in SPECIAL)]
+        result = super()._read_group(domain, groupby, new_aggregates, having, offset, limit, order)
```

Comment:
The former name was chosen on purpose: `new_result` corresponded to `new_aggregates`, while `result` corresponded to `aggregates`.

Reply:
Ok I see, I made this change because we are looping on `result` instead of `new_result` just setting `result = []`.
Then I am currently fixing the issue without change naming.